src/analyzing/AST_to_IL.ml

```diff
@@ -880,16 +880,66 @@ and expr_aux env ?(void = false) g_expr =
           in
           expr env last)
   | G.Record fields -> record env fields
-  | G.Container (G.Dict, xs) -> dict env xs g_expr
+  | G.Container (G.Dict, xs) -> (
+      (* Lua-specific: Check if this is a Lua array-like table with NextArrayIndex *)
+      let l, entries, r = xs in
+      let is_lua_array_table =
+        env.lang =*= Lang.Lua &&
+        List.for_all
+          (fun entry ->
+            match entry.G.e with
+            | G.Assign ({ e = IdSpecial (NextArrayIndex, _); _ }, _, _) -> true
+            | _ -> false)
+          entries
+      in
+      if is_lua_array_table then
+        (* Extract values from Assign(NextArrayIndex, value) and treat as array *)
+        let values =
+          entries
+          |> List.map (fun entry ->
+                 match entry.G.e with
+                 | G.Assign (_, _, value) -> value
+                 | _ -> assert false)
+        in
+        let env, vs = List.fold_left_map (fun env v -> expr env v) env values in
+        (env, mk_e (Composite (CList, (l, vs, r))) eorig)
+      else
+        (* Regular dict *)
+        dict env xs g_expr)
   | G.Container (kind, xs) ->
       let l, xs, r = xs in
       let env, xs = List.fold_left_map (fun env x -> expr env x) env xs in
       let kind = composite_kind ~g_expr kind in
       (env, mk_e (Composite (kind, (l, xs, r))) eorig)
   | G.Comprehension _ -> todo env.stmts (G.E g_expr)
-  | G.Lambda fdef ->
-      let lval = fresh_lval (snd fdef.fkind) in
-      let _, fdef =
+  | G.Lambda fdef -> (
+      (* Elixir implicit parameter lambda conversion:
+       * Elixir lambdas like `fn x -> sink(x) end` are parsed with an implicit parameter
+       * and a Switch body. Convert them to explicit parameter lambdas. *)
+      let converted_fdef =
+        match (env.lang, Tok.unbracket fdef.fparams, fdef.fbody) with
+        | ( Lang.Elixir,
+            [ G.Param { pname = Some (param_name, _); _ } ],
+            G.FBStmt { s = G.Switch (_, Some (G.Cond _), cases); _ } )
+          when param_name =*= "!_implicit_param!" && List.length cases =*= 1 -> (
```

Comment:
I recommend to not hardcode `!_implicit_param!`, this will change soon.

Is there a way to avoid depending on this string? All Elixir `fn` have this encoding anyway! Maybe `List.length` is enough,

Reply:
As a side comment, knowing of course you have to do this here to catch taints, this kind of thing introduces tech debt....

Whenever the elixir translation changes, this also has to change! And it might be forgotten....

Not saying not to keep it, but it's one of the reasons I'd rather see that stuff in a separated module that we can "remember" to update somehow.